train/compute/python/tools/trace_link.py

```diff
@@ -0,0 +1,309 @@
+import json
+from param_bench.train.compute.python.tools.execution_graph import ExecutionGraph
+import sys
+
+
+# Increase recursion limit
+sys.setrecursionlimit(10**6)
+
+
+def read_dictionary_from_json(file_path):
+    with open(file_path, 'r') as file:
+        data = json.load(file)
+        return data
+
+
+# Add and sort ET nodes from the execution graph  
+def collect_nodes(node):
+    def traverse(node):
+        nonlocal nodes
+        nodes.append(node)
+        for child in node.children:
+            traverse(child)
+
+    nodes = []
+    traverse(node)
+    sorted_nodes = sorted(nodes, key=lambda x: x.id)
+    return sorted_nodes
+
+
+class Kineto_node:
+    def __init__(self, name, start, end, id):
+        self.name = name
+        self.start = start
+        self.end = end
+        self.id = id
+        self.children = []
+
+
+# Function to transform your self-defined tree to a directed graph with max depth
+def transform_to_graph_depth(node, max_depth=100):
+    graph = nx.DiGraph()
+    add_node_to_graph_depth(node, graph, max_depth, 1)
+    return graph
+
+
+# Helper function to recursively add nodes and edges to the graph with max depth
+def add_node_to_graph_depth(node, graph, max_depth, cur_depth):
+    graph.add_node(node.id, label=node.name)
+    if cur_depth == max_depth:
+        return
+    for child in node.children:
+        graph.add_node(node.id, label=node.name)
+        add_node_to_graph_depth(child, graph, max_depth, cur_depth + 1)
+
+
+# Custom node comparison function for edit distance
+def node_compare(n1, n2):
+    return n1 == n2
+
+
+# Find the segment that has a length closest to the target
+def find_closest_segment(segs, target_length):
+    closest_length = float('inf')
+    closest_seg = None
+
+    for seg in segs:
+        length_difference = abs(len(seg) - target_length)
+        if length_difference < closest_length:
+            closest_length = length_difference
+            closest_seg = seg
+
+    return closest_seg
+
+
+def exact_match(kineto_et_events, et_nodes):
+    # Since kineto trace is missing the annotations for processes/threads, we add them back to match with ET
+    kineto_event_per_thread = {}
+
+    process_end_time = -1
+    for i in range(len(kineto_et_events)):
+        event = kineto_et_events[i]
+        if event['tid'] not in kineto_event_per_thread:
+            kineto_event_per_thread[event['tid']] = {}
+            kineto_event_per_thread[event['tid']]['ts'] = event['ts']
+            kineto_event_per_thread[event['tid']]['dur'] = event['ts'] + event['dur']
+            kineto_event_per_thread[event['tid']]['index'] = i
+        else:
+            kineto_event_per_thread[event['tid']]['dur'] = max(kineto_event_per_thread[event['tid']]['dur'], event['ts'] + event['dur'])
+        process_end_time = max(process_end_time, event['ts'] + event['dur'])
+
+    process_event = {'name': '[pytorch|profiler|execution_graph|process]', 'ts': kineto_et_events[0]['ts'], 
```

Comment:
These event strings need to be updated due to the latest changes from "execution_graph" to "execution_trace". See PR https://github.com/pytorch/pytorch/pull/99694

Reply:
Maybe make these strings global constants if they are used in multiple places.